pretyping/locusops.ml

```diff
@@ -27,12 +28,43 @@ let occurrences_map f = function
     if l' = [] then AllOccurrences else AllOccurrencesBut l'
   | (NoOccurrences|AllOccurrences|AtLeastOneOccurrence) as o -> o
 
-let convert_occs = function
-  | AtLeastOneOccurrence -> (false,[])
-  | AllOccurrences -> (false,[])
-  | AllOccurrencesBut l -> (false,l)
-  | NoOccurrences -> (true,[])
-  | OnlyOccurrences l -> (true,l)
+type occurrences_count = (int * int list) * (bool * int)
```

Comment:
Given that the type is abstract, a record wouldn't hurt in the ml file.

Reply:
Three records you mean? Will do if you like it better.

Reply:
WDYM? I was just asking to turn `occurrences_count` into a 4-field record, so that we understand what argument is supposed to be.

Reply:
I mean that from some point of view, this is a pair/record of two pairs/records.

The right part is fixed (and readonly), so it does not need to be copied as two distinct pointers each time we build a new record. The left part is the part which is updated.

Not that this distinction is important in practice, but a 4-field tuple would not highlight this invariant (though we can maybe also choose a name which highlight this).

Reply:
Well, then you can use three fields, and keep the RHS in a pair.

Reply:
Ok, done.